File: sdks/dotnet/src/Models/PaginatedResponse.cs
using System.Collections.Generic;
using Newtonsoft.Json;

namespace Puxbay.SDK.Models
{
    public class PaginatedResponse<T>
    {
        [JsonProperty("count")]
        public int Count { get; set; }

        [JsonProperty("next")]
        public string Next { get; set; }

        [JsonProperty("previous")]
        public string Previous { get; set; }

        [JsonProperty("results")]
        public List<T> Results { get; set; }
    }
}

File: sdks/php/src/Models/PaginatedResponse.php
<?php

namespace Puxbay\Models;

class PaginatedResponse
{
    public function __construct(
        public readonly int $count,
        public readonly ?string $next,
        public readonly ?string $previous,
        public readonly array $results,
    ) {
    }

    public static function fromArray(array $data, callable $itemFactory): self
    {
        return new self(
            count: (int) $data['count'],
            next: $data['next'] ?? null,
            previous: $data['previous'] ?? null,
            results: array_map($itemFactory, $data['results'] ?? []),
        );
    }
}
